pkg/scheduler/scheduler.go

```diff
@@ -351,6 +357,27 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 	return nil
 }
 
+func (s *Scheduler) applyAdmissionWithSSA(ctx context.Context, w *kueue.Workload) error {
+	return s.client.Patch(ctx, w, client.Apply, client.FieldOwner(constants.AdmissionName))
+}
+
+// workloadAdmissionFrom returns only the fields necessary for admission using
+// ServerSideApply.
+func workloadAdmissionFrom(w *kueue.Workload) *kueue.Workload {
+	return &kueue.Workload{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:        w.UID,
+			Name:       w.Name,
+			Namespace:  w.Namespace,
+			Generation: w.Generation, // Produce a conflict if there was a change in the spec.
```

Comment:
but then what conflicts does server-side apply help with vs Update? I assume changes in Status doesn't produce conflicts with the Update, or does it?

Reply:
Status changes do produce conflicts with Update (at least I have observed them in integration tests).

Reply:
I'm not sure I understand this, but that sounds like something we might have to be careful about. Can someone explain that to me?

Reply:
We want the Apply to fail if there was any chance in the spec with respect to the version we took a decision on. However, we don't care if the status has changed.

That was our motivation to use SSA. Daniel Smith suggested we use the generation for this purpose https://github.com/kubernetes-sigs/kueue/issues/164#issuecomment-1083394946